var/spack/repos/builtin/packages/py-awkward/package.py

```diff
@@ -31,13 +43,19 @@ class PyAwkward(PythonPackage):
     patch("pybind11.patch", when="@:1.2.2")
     patch("pybind11_02.patch", when="@1.2.3:1.8.0")
 
-    depends_on("py-setuptools@42.0:", type=("build", "run"))
+    depends_on("py-setuptools@42.0:", when="@:1", type=("build", "run"))
     depends_on("py-pyyaml", type="build")
+    depends_on("py-hatchling@1.10:", when="@2:", type="build")
+    depends_on("py-hatch-fancy-pypi-readme", when="@2:", type="build")
 
```

Comment:
> My vote is for a separate py-awkward-cpp package and pinning the dep version in each version of py-awkward.

This would then end up with something like this:
```suggestion

    _awkward_to_awkward_cpp_map = [
      ("@2.0.0", "@2"),
      ("@2.0.1", "@3"),
      ("@2.0.2", "@4"),
      ("@2.0.3:2.0.4", "@5"),
      ("@2.0.5", "@6"),
      ("@2.0.6", "@7"),
      ("@2.0.7", "@8"),
      ("@2.0.8", "@9"),
    ]
    for _awkward, _awkward_cpp in _awkward_to_awkward_cpp_map:
        depends_on("py-awkward-cpp{}".format(_awkward_cpp), when=_awkward, type=("build", "run"))

```

Reply:
Yep. We do the same thing in a lot of packages. I agree, it's annoying. You don't _have_ to add all of those versions if you don't want to, you could only add the latest version.